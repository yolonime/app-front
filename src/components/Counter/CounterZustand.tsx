import { create } from "zustand";

type State = {
  counter: number;
  start: boolean;
  increase: () => void;
  decrease: () => void;
};

const useCounterStore = create<State>((set) => ({
  counter: 0,
  start: false,
  increase: () => set((state) => ({ counter: state.counter + 1, start: true })),
  decrease: () =>
    set((state) => ({
      counter: state.counter - 1,
      start: true,
    })),
}));

const Counter2 = () => {
  const counter = useCounterStore((state) => state.counter);
  const start = useCounterStore((state) => state.start);
  const increase = useCounterStore((state) => state.increase);
  const decrease = useCounterStore((state) => state.decrease);

  console.log(start);

  return (
    <div className="flex flex-col">
      <span className="text-[48px]">{counter}</span>
      <div>
        <button className="text-xl" onClick={() => increase()}>
          Add
        </button>
        <button className="text-xl ml-2" onClick={() => decrease()}>
          -
        </button>
      </div>
    </div>
  );
};

export default Counter2;
