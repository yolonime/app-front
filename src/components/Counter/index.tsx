import { MouseEventHandler, useReducer } from "react";
import { DECREMENT, INCREMENT } from "../../store/actions";
import { reducer, State } from "../../store/reducer/counter";

const INITIAL_STATE: State = { count: 0 };

const Counter = () => {
  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);

  const add: MouseEventHandler = () => {
    dispatch({ type: INCREMENT });
  };

  const substract: MouseEventHandler = () => {
    dispatch({ type: DECREMENT });
  };

  return (
    <div className="flex flex-col">
      <span className="text-[48px]">{state.count}</span>
      <div>
        <button className="text-xl" onClick={add}>
          Add
        </button>
        <button className="text-xl ml-2" onClick={substract}>
          -
        </button>
      </div>
    </div>
  );
};

export default Counter;
