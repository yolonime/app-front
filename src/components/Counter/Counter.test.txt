import { render, screen } from "@testing-library/react";
import Counter from ".";
import userEvent from "@testing-library/user-event";

const renderComponent = () => {
  return render(<Counter />);
};

describe("Counter component", () => {
  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it("should display 0 at init", () => {
    renderComponent();

    const value = screen.getByText("0");

    expect(value).toBeVisible();
  });

  it("should increment value on Add button click", async () => {
    renderComponent();

    const button = screen.getByRole("button", { name: "Add" });

    await userEvent.click(button);

    expect(screen.getByText("1")).toBeVisible();
  });

  it("should decrement value on - button click", () => {
    renderComponent();

    const button = screen.getByRole("button", { name: "-" });

    userEvent.click(button);

    expect(screen.getByText("-1")).toBeVisible();
  });
});
