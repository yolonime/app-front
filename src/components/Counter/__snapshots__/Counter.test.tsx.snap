// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render correctly 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <span
      class="text-[48px]"
    >
      0
    </span>
    <div>
      <button
        class="text-xl"
      >
        Add
      </button>
      <button
        class="text-xl ml-2"
      >
        -
      </button>
    </div>
  </div>
</div>
`;
